### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by mathematicalmichael**
doing this on top of the plotting examples assuming it'll be merged after #302

### CODE REVIEW

**Comment by mathematicalmichael** [comment]
Things I'd like to "improve"
- Allow markersize as argument to scatter. Large(ish) N are hard to see with `markersize=75` fixed.
- Labeling for scatterplot uses x/y and surfaces uses lambda. No attention is paid to input/output distinction. We should stick to lambda by default.
- If available, strings for column names (either QoI or Params) used to label axes in plots.

**Comment by mathematicalmichael** [comment]
- just noticed `scatter_2D_output` saves with the wrong default filename. fixed.

**Comment by mathematicalmichael** [comment]
the problem of the q1_q1 scatter plots came from `plotD.show_data_domain_multi` not excluding the comparative dimension in `for i in Q_nums:`

**Comment by mathematicalmichael** [comment]
`plotD.scatter_rhoD` had very unexpected behavior with the passing of `showdim`, addressed by switching order in which conditions are checked. (if you passed 3D array and showdim=2, it would ignore the fact that you want 2D scatterplots). fixed. 
Same function... ref_sample was not being indexed when passed to scatterplot. fixed.

**Comment by mathematicalmichael** [comment]
had to make some changes in `PlotDomains` that uncovered some failing tests (weren't caught because the code coverage never checked these corner cases). `scatter_rhoD` was the problem, and the way that the "if/then" statements were being checked combined with the dimensions of the tests led to an avoidance of this problem (`showdim` not `None` paired with `param_ref` not `None`). I added the logical conditions to catch these corner cases, but I don't think it's worth writing the extra tests.

**Comment by smattis** [comment]
@mathematicalmichael I can do it this week, probably tomorrow. I didn't realize that it was ready yet until now.

File: test/test_postProcess/test_plotDomains.py [hunk 1]
```diff
@@ -222,7 +222,14 @@ def check_show_param(self, samples, sample_nos, p_ref, save, lnums,
             plotDomains.scatter_rhoD(disc_obj_temp, p_ref, sample_nos, 'input',
                     self.rho_D, lnums, None, showdim, save, False)
             go = True
-        except (RuntimeError, TypeError, NameError):
+        except (RuntimeError, TypeError, NameError) as error:
```
Comment (smattis) [inline test/test_postProcess/test_plotDomains.py]: Do we need these print statements in the tests?

File: test/test_postProcess/test_plotDomains.py [hunk 2]
```diff
@@ -266,12 +273,12 @@ def check_show_data(self, data, sample_nos, q_ref, save, qnums, showdim):
             go = True
         except (RuntimeError, TypeError, NameError):
             print("ERROR")
-            print(data.shape)
-            print(q_ref)
-            print(sample_nos)
-            print(save)
-            print(qnums)
-            print(showdim)
+            print("data shape:", data.shape)
```
Comment (smattis) [inline test/test_postProcess/test_plotDomains.py]: Do we need these print statements in the tests?

Comment (mathematicalmichael) [reply test/test_postProcess/test_plotDomains.py]: some tests had been printing this kind of info, some were not. If a test fails, it is helpful to have something to read, and I thought the lack of this printing in all the tests was just an overlooked "to do" 
A lot of time was spent debugging plotting code due to the way these tests were written to not catch error printouts. When a test fails, it is completely unclear which case it was.

**Comment by mathematicalmichael** [comment]
> I would prefer if in the `examples` directory, you added a `templates` subdirectory and move the jupyter notebook template there.
> Also, make a subdirectory for the plotting examples. I would prefer not to have lone files in the `examples` directory.
> 
> I really like the notebooks in general. There is an extra line of whitespace at the end of most of them. It would look slightly better if this was removed.
> 
> Otherwise, these things and my inline comments are all that need to be addressed.

Will do, but the line at the end of the notebooks occurs when you "Run All" and it's really hard to avoid having it there. But you're right. Aesthetically, it's more pleasing without.

File: test/test_postProcess/test_plotDomains.py (hunk 1 above)
Comment (mathematicalmichael) [reply test/test_postProcess/test_plotDomains.py]: some tests had it, some did not. I liked the tests that had it, so I added it to places I encountered problems.

Comment (smattis) [reply test/test_postProcess/test_plotDomains.py]: Ok. Keep them. I get it.

File: test/test_postProcess/test_plotDomains.py (hunk 2 above)
Comment (smattis) [reply test/test_postProcess/test_plotDomains.py]: Ok. Keep them. I get it.

**Comment by mathematicalmichael** [comment]
I like the `/templates` subdirectory suggestion and can actually see some good files to throw in there. For example, parallel_and_serial_sampling doesn't quite run the same way as other examples, nor does `fromFile_ADCIRCMap`, `FEniCS`, and `matfiles`. Thoughts about moving those into `templates`?

**Comment by smattis** [comment]
I don't think those really belong in `templates`. They actually all are working examples. They just aren't put in notebook form, and it would be pretty difficult or impossible to do so. Actually these examples, are really useful for figuring out how to use the code for a practical problem. I would leave them where they are for now.

Actually, a nice thing to add to the Readme would be links to the documented examples with the notebooks, so it is clear where those examples are.

**Comment by mathematicalmichael** [comment]
@smattis do you mean like, this kind of link? (./examples/contaminantTransport/contaminant.ipynb)

[Contaminant Transport](https://github.com/mathematicalmichael/BET/blob/improvements/examples/contaminantTransport/contaminant.ipynb)

**Comment by smattis** [comment]
Yeah. That type of link.

**Comment by mathematicalmichael** [comment]
> Yeah. That type of link.

Done, double checked that they worked in the README preview on my branch. They're much more boring without saved images, but oh well. Maybe later on I will split the plotting examples up (1D vs 2D marginals) and use those as a place to demonstrate some pretty pictures. Overall though... it's a hard choice between committing clean empty notebooks and rich ones that are a larger size/confusing-ish to git (simply changing the color scheme in a plot may be interpreted as a re-write of the notebook).

**Comment by smattis** [comment]
@mathematicalmichael  Looks good to me. I am merging now.
